src/main/java/uk/gov/gchq/palisade/resource/AbstractLeafResource.java

```diff
@@ -33,13 +34,13 @@
  * See {@link FileResource} for a concrete implementation with an id.
  * This class is mostly used when deserialisation to a LeafResource is required, but the interface can't be used.
  */
-public abstract class AbstractLeafResource extends AbstractResource implements LeafResource, ChildResource {
+public abstract class AbstractLeafResource extends AbstractResource implements LeafResource {
 
     private String type;
     private String serialisedFormat;
     private ConnectionDetail connectionDetail;
     private ParentResource parent;
-    private Map<String, Object> attributes = new HashMap<>();
+    private HashMap<String, Serializable> attributes = new HashMap<>();
```

Comment:
Just a double check, does this have to be a HashMap?  Can't it be a Map?

Reply:
I tried, and it pretty much has to be...

Java doesn't support traits or join-types, so there's no way to specify this class requires an object T such that T extends Map and Serializable.
My thought was you could declare a 'dummy' `interface SerializableMap<K, V> extends Serializable, Map<K, V> {}`, but then Java decides it's illegal to cast a HashMap to this type (suppose we were to add a single method to that interface...)

There are "better" ways around it, but not worth the effort (a SerializableMap object would have to wrap a Serializable Map and delegate all map methods to this concrete implementation).

I think weighing the options, HashMaps are almost always the go-to choice, so I don't think its too restrictive to say we are using it over a more generic interface.